### PR TITLE
Add org.kde.kstars

### PR DESCRIPTION
**Pull request by eszlari**
Pinging upstream: @knro

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.kde.kstars

**Comment by nedrichards** [comment]
bot, build org.kde.kstars

**Comment by knro** [comment]
That's awesome, thanks! But in my last discussion, I knew that udev is not supported in flatbak. INDI uses udev to grant permission to USB devices and to upload firmware (that's the only way I know of).

File: org.kde.kstars.yaml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      - type: archive
+        url: https://github.com/frankosterfeld/qtkeychain/archive/v0.9.1.tar.gz
+        sha256: 9c2762d9d0759a65cdb80106d547db83c6e9fdea66f1973c6e9014f867c6f28e
+
+  - name: kstars
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DCMAKE_C_FLAGS='-lcurl'
```
[the remark this answers is not in the document]
Comment (eszlari) [reply org.kde.kstars.yaml]: It seems so:
```
[588/588] Linking CXX executable kstars/kstars
FAILED: kstars/kstars 
: && /run/ccache/bin/c++   -std=c++0x -fno-operator-names -Wall -Wextra -Wcast-align -Wchar-subscripts -Wformat-security -Wno-long-long -Wpointer-arith -Wundef -Wnon-virtual-dtor -Woverloaded-virtual -Werror=return-type -Wvla -Wdate-time -fdiagnostics-color=always -fexceptions -Wno-non-virtual-dtor -fdata-sections -ffunction-sections -D_FORTIFY_SOURCE=2 -fstack-protector-all -Wcast-align -fPIE -Wa,--noexecstack  -Wall -Wextra -Wno-unused-but-set-variable -g -O2 -g -DNDEBUG  -Wl,--enable-new-dtags -L/app/lib -Wl,--gc-sections  -Wl,-z,nodump -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -pie kstars/CMakeFiles/kstars.dir/main.cpp.o kstars/CMakeFiles/kstars.dir/qrc_kstars.cpp.o kstars/CMakeFiles/kstars.dir/kstars_automoc.cpp.o  -o kstars/kstars  -rdynamic kstars/libKStarsLib.a datahandlers/libLibKSDataHandlers.a kstars/htmesh/libhtmesh.a /usr/lib/libKF5Crash.so.5.50.0 /usr/lib/libKF5NewStuff.so.5.50.0 /usr/lib/libKF5NewStuffCore.so.5.50.0 /usr/lib/libKF5KIOFileWidgets.so.5.50.0 /usr/lib/libKF5Bookmarks.so.5.50.0 /usr/lib/libKF5XmlGui.so.5.50.0 /usr/lib/libKF5Attica.so.5.50.0 /usr/lib/libKF5ItemViews.so.5.50.0 /usr/lib/libKF5Solid.so.5.50.0 /usr/lib/libKF5Plotting.so.5.50.0 /usr/lib/libKF5Notifications.so.5.50.0 /usr/lib/libQt5PrintSupport.so.5.11.1 /usr/lib/libQt5Sql.so.5.11.1 /usr/lib/libQt5Svg.so.5.11.1 /usr/lib/libQt5Quick.so.5.11.1 /usr/lib/libQt5Qml.so.5.11.1 /usr/lib/libQt5WebSockets.so.5.11.1 -lz /app/lib/libqt5keychain.so.0.9.1 /usr/lib/libKF5NotifyConfig.so.5.50.0 /usr/lib/libKF5KIOWidgets.so.5.50.0 /usr/lib/libKF5ConfigWidgets.so.5.50.0 /usr/lib/libKF5Codecs.so.5.50.0 /usr/lib/libKF5KIOCore.so.5.50.0 /usr/lib/libQt5Network.so.5.11.1 /usr/lib/libQt5Concurrent.so.5.11.1 /usr/lib/libKF5Service.so.5.50.0 /usr/lib/libKF5I18n.so.5.50.0 /usr/lib/libKF5Auth.so.5.50.0 /usr/lib/libKF5JobWidgets.so.5.50.0 /usr/lib/libKF5CoreAddons.so.5.50.0 /usr/lib/libKF5Completion.so.5.50.0 /usr/lib/libKF5WidgetsAddons.so.5.50.0 /usr/lib/libQt5Widgets.so.5.11.1 /usr/lib/libKF5ConfigGui.so.5.50.0 /usr/lib/libQt5Gui.so.5.11.1 /usr/lib/libQt5Xml.so.5.11.1 /usr/lib/libKF5ConfigCore.so.5.50.0 /usr/lib/libQt5DBus.so.5.11.1 /usr/lib/libQt5Core.so.5.11.1 -lm -Wl,-Bstatic -lcfitsio -Wl,-Bdynamic -lgsl -lgslcblas -lpthread -Wl,-Bstatic -lindiclient -Wl,-Bdynamic -lnova -lz -lwcs -lraw && :
/app/lib/libcfitsio.a(cfileio.o): In function `ffihtps':
/run/build/cfitsio/cfileio.c:7590: undefined reference to `curl_global_init'
/app/lib/libcfitsio.a(cfileio.o): In function `ffchtps':
/run/build/cfitsio/cfileio.c:7603: undefined reference to `curl_global_cleanup'
/app/lib/libcfitsio.a(drvrnet.o): In function `https_open_network':
/run/build/cfitsio/drvrnet.c:1212: undefined reference to `curl_easy_init'
/run/build/cfitsio/drvrnet.c:1214: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1221: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1223: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1224: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1226: undefined reference to `curl_easy_setopt'
/app/lib/libcfitsio.a(drvrnet.o):/run/build/cfitsio/drvrnet.c:1230: more undefined references to `curl_easy_setopt' follow
/app/lib/libcfitsio.a(drvrnet.o): In function `https_open_network':
/run/build/cfitsio/drvrnet.c:1250: undefined reference to `curl_easy_perform'
/run/build/cfitsio/drvrnet.c:1352: undefined reference to `curl_easy_cleanup'
/run/build/cfitsio/drvrnet.c:1274: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1275: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1276: undefined reference to `curl_easy_perform'
/run/build/cfitsio/drvrnet.c:1345: undefined reference to `curl_easy_cleanup'
/run/build/cfitsio/drvrnet.c:1285: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1286: undefined reference to `curl_easy_perform'
/run/build/cfitsio/drvrnet.c:1326: undefined reference to `curl_easy_setopt'
/run/build/cfitsio/drvrnet.c:1327: undefined reference to `curl_easy_perform'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
Error: module kstars: Child process exited with code 1
```

**Comment by eszlari** [comment]
> But in my last discussion, I knew that udev is not supported in flatbak. INDI uses udev to grant permission to USB devices and to upload firmware (that's the only way I know of).

So this a blocker, unless you want to offer a version on Flathub without the full feature set.

**Comment by knro** [comment]
Yes it is a blocker since users expect KStars+INDI on Linux to be fully working.

**Comment by z3ntu** [comment]
Well I seem to be a bit late with publishing the manifest that I created for kstars: https://github.com/z3ntu/flathub/commit/5620fd4ed854c08c52d2f8525548b90085e39ce6

File: org.kde.kstars.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (z3ntu) [reply org.kde.kstars.yaml]: This is another issue, see https://github.com/z3ntu/flathub/commit/5620fd4ed854c08c52d2f8525548b90085e39ce6#diff-397c3d6647204152c94321b10db3e879R44

**Comment by barthalion** [comment]
I don't think we can do much if @knro wants full functionality – there is no way to export any udev rules (do I understand correctly how it works?). One way to proceed would be to disable some functionality basing on presence of `/.flatpak-info` file and show a prompt to documentation (even wiki page of the future repo) what has to be done on host system to make INDI work. Would it be possible? If not, I don't think we can merge it.

**Comment by knro** [comment]
Absolutely. If udev rules cannot be somehow exported, then this is a no-go. It would be a support nightmare once the questions starting pouring in about why isn't this working as expected. I would love for this work, but the current limitation is unfortunately a show stopper.

**Comment by eszlari** [comment]
@knro 

> INDI uses udev to grant permission to USB devices and to upload firmware (that's the only way I know of).

For the firmware: What about fwupd? https://fwupd.org

**Comment by knro** [comment]
I don't know, we use fxload to upload the firmware. Also, what about USB device permission? We need to grant 0666 so that regular users can access it.

**Comment by eszlari** [comment]
I don't know the technical details, but there are already several apps on Flathub that utilize libusb for accessing to USB devices.

**Pull request closed without merging**